src/screens/ShareTrip/ShareTrip.tsx

```diff
@@ -56,48 +186,69 @@ export default function ShareTrip(props: Props): JSX.Element {
                 <SvgLogo style={styles.logo} height={150} width={150} />
-                <Button
-                    containerStyle={styles.btnContainer}
-                    buttonStyle={styles.shareBtn}
-                    title={t("screens.shareTrip.share")}
-                    titleStyle={styles.btnTextStyle}
-                    onPress={() => handleShare(getInvitationLinkPromise(), t)}
-                    loading={loading}
-                />
+                {mode === Mode.SHARE_TRIP && (
+                    <Button
+                        containerStyle={styles.btnContainer}
+                        buttonStyle={styles.shareBtn}
+                        title={t("screens.shareTrip.share")}
+                        titleStyle={styles.btnTextStyle}
+                        onPress={() => handleSubmitButton()}
+                        loading={loadingCreateInvitation}
+                    />
+                )}
                 <Button
                     containerStyle={styles.btnContainer}
                     buttonStyle={styles.planTripBtn}
-                    title={t("screens.shareTrip.planTrip")}
+                    title={t(
+                        mode === Mode.SHARE_TRIP
+                            ? "screens.shareTrip.planTrip"
+                            : "screens.shareTrip.joinTrip"
+                    )}
                     titleStyle={styles.btnTextStyle}
-                    onPress={() =>
-                        props.navigation.reset({
-                            index: 1,
-                            routes: [
-                                {
-                                    name: Routes.DASHBOARD,
-                                },
-                                {
-                                    name: Routes.ITINERARY,
-                                    params: {
-                                        screen: Routes.ITINERARY,
-                                        params: {
-                                            tripId: trip.id,
-                                            tripName: trip.name,
-                                        },
+                    // TODO - Redirect to trip itinerary screen
+                    onPress={() => {
+                        if (mode === Mode.JOIN_TRIP && invitation) {
+                            executeJoinTripMutation({
+                                variables: {
+                                    input: {
+                                        tripId: trip.trip.id,
+                                        invitationId: invitation.invitation.id,
                                     },
                                 },
-                            ],
-                        })
-                    }
+                            })
+                                .then(() => handlePlanTripNavigation())
+                                .catch((e) => {
+                                    console.error(e);
```

Comment:
Isn't this a case where we should properly notify the user with a `Toast`?

Reply:
@valentingc isn't it automatically called?

Reply:
Yep, should come automatically afaik